zperf: Add support for nRF7002DK

krish2718 opened the pull request:
Add ability to run Zperf sample on nRF70 Series for benchmarking, also add a generic Wi-Fi overlay that uses hostap as NM, can be used by other boards as well.

See also https://github.com/zephyrproject-rtos/zephyr/issues/80111#issuecomment-2563041272

krish2718 committed:
net: l2: wifi: Fix PHY rate display

The WPA supplicant already converts this to Mbps.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
samples: net: zperf: Add nRF7002DK support

Add settings to get peak throughputs for nRF7002DK, this relies on the
overlay-wifi-nm-hostap.conf.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
samples: net: zperf: Use highest frequency for nRF5340

When running Wi-Fi benchmarks on nRF7002DK use the highest CPU
frequency 128MHz.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

pdgendt commented on samples/net/zperf/src/main.c:
```diff
@@ -41,6 +47,21 @@ static int enable_usb_device_next(void)
 int main(void)
 {
+
+#if defined(CONFIG_SOC_NRF5340_CPUAPP)
```
Not a big fan of having SoC specific code directly in the sample's `main`. Maybe add a separate C file and use `SYS_INIT`. The file can then be conditionally added in CMake with `target_sources_ifdef`.

krish2718 replied:
Agreed, I did raise a Nordic internal request to provide the CPU frequency configuration support in a DTS and other abandoned avenues (https://github.com/zephyrproject-rtos/zephyr/pull/42935).

I saw this was already being used in https://github.com/zephyrproject-rtos/zephyr/blob/ee250a11ce503fdf24793afac0712bacd26759da/subsys/bluetooth/audio/shell/bap_usb.c#L554 hence used the same (just had to use `#if` to avoid undefined references).

jukkar replied:
I agree with Pieter, please no hw specific checks in generic sources. If really need we could perhaps have a generic kconfig option that is selected by hw code.

krish2718 replied:
There isn't such Kconfig/DTS option, I will move it to a separate file and handle it in the cmake.

rlubos commented on samples/net/zperf/overlay-wifi-nm-hostap.conf:
```diff
@@ -0,0 +1,19 @@
+# For Wi-Fi STA mode
```
Couldn't we just use the Wi-Fi snippet used by other samples?
https://github.com/zephyrproject-rtos/zephyr/blob/main/snippets/wifi-ipv4/wifi-ipv4.conf
There're slight differences in the configuration, but if we need more resources for this sample we could perhaps adjust it.

krish2718 replied:
Ah right, forgot about this snippet, we can use that, thanks.

krish2718 replied:
```
+# Zperf TCP=6, UDP=2, Others=1
+# WPA supplicant: Global ctrl_iface: 2, per interface: 2
+CONFIG_ZVFS_OPEN_MAX=20
+CONFIG_NET_SOCKETS_POLL_MAX=20
+CONFIG_ZVFS_POLL_MAX=20
```
How do we deal with these options? I can move to board specific, but these are common to wi-fi but only need when wi-fi and zperf are used together?

rlubos replied:
The snippet already sets `CONFIG_ZVFS_OPEN_MAX=24`. As for `CONFIG_ZVFS_POLL_MAX`, snippet sets this to 9, same as the default `prj.conf` in the zperf sample. Do we really need to have this set to 20? I'm not sure if I understand why would we need more than the default from prj.conf specifically for Wi-Fi case. But if that really the case - I don't see a problem with increasing the config value in the snippet.

Regarding `CONFIG_NET_SOCKETS_POLL_MAX` - it's deprecated and redundant with `CONFIG_ZVFS_POLL_MAX`, we should just drop it.

krish2718 replied:
I have dropped that commit, I will run more tests with snippet and get back to you on the socket settings. Thanks.

krish2718 replied:
The snippet works as is, no need to update, thanks.

krish2718 replied:
I had to update the snippet too, earlier I had the increased settings in the board file.

rlubos reviewed (commented):
We could still update the readme file, mentioning it can be run with the Wi-Fi snippet, like some other samples do:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/sockets/http_get/README.rst#wi-fi

krish2718 committed:
snippets: wifi-ipv4: Increase max polling sockets

When using this snippet with Zperf a boot warning is seen, fix the
warning by increasing the sockets.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>
